src/software/ai/motion_constraint/motion_constraint_visitor.cpp

```diff
@@ -2,6 +2,8 @@
 
 #include "software/ai/hl/stp/tactic/all_tactics.h"
 
+void MotionConstraintVisitor::visit(const Tactic &tactic) {}
+
```

Comment:
we shouldn't have this visitor here because we want to explicitly define these visitors and have the compiler complain if we mess it up

Reply:
bump @williamckha

Reply:
We can't forward declare nested classes (e.g. `ShootSkill::SkillTactic`) in the TacticVistor file so I will have to think about how to solve this

Reply:
I have removed the nested `SkillTactic` class from `BaseSkill` and added a `AssignedSkillTactic<Skill>` that we can use to execute skills

Reply:
okay, this means we can remove this visitor impl for `Tactic`?